dbt/tests/unit_tests/test_weighted_quantile.sql

```diff
@@ -1,31 +1,57 @@
+{{ config(tags = ['unit','weighted_quantile']) }}
+
 WITH test_data AS (
     -- Mock dataset for testing
     SELECT * FROM (VALUES
-        (10, 2),
-        (20, 3),
-        (30, 5),
-        (40, 10),
-        (50, 10)
-    ) AS t(column_name, weight_col)
+        ( 200, NULL, 0),
+        (NULL,  200, 0),
+        (  10,    2, 0),
+        (  20,    3, 0),
+        (  30,    5, 0),
+        (  40,   10, 1),
+        (  50,   10, 1),
+        ( 100, NULL, 1),
+        (NULL,  100, 1)
+    ) AS t(column_name, weight_col, selector)
```

Comment:
Keep original 10-20-30-40-50 but add NULL entries, and add selector column to test row conditions